ccm_web/server/src/api/api.service.ts

```diff
@@ -142,10 +146,12 @@ export class APIService {
     const requestor = await this.canvasService.createRequestorForUser(user, '/api/v1/')
     const NS_PER_SEC = BigInt(1e9)
     const start = process.hrtime.bigint()
-    const apiPromises = sectionIds.map(async (si) => {
-      const sectionHandler = new SectionApiHandler(requestor, si)
-      return await sectionHandler.unmergeSection()
-    })
+    const apiPromises = createLimitedPromises<CanvasCourseSectionBase | APIErrorData>(
+      sectionIds.map(si => async () => {
+        const sectionHandler = new SectionApiHandler(requestor, si)
+        return await sectionHandler.unmergeSection()
+      })
+    )
```

Comment:
See https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/327/files#r803102325, same logic applies.

Reply:
Why unmerge has to be wrapped with the concurrency call? since we mostly ever going to do 1 section unmerge at a time

Reply:
That's fine, I can remove it if you want, I was just looking for feedback on which to remove it from.

Reply:
Theoretically, someone could do up to 250 using the API, but not likely to ever happen.

Reply:
Since UI doesn't support bulk unmerge and only one at a time, I believe we can remove it

Reply:
So you're not worried about direct API calls? This is the DTO for both merge and unmerge: https://github.com/tl-its-umich-edu/canvas-course-manager-next/blob/main/ccm_web/server/src/api/dtos/api.section.ids.dto.ts

Reply:
Ok I checked again, even though this is list ` sectionIds.map` but that list will always have 1 section for unmerge section. So the wrapper for p-limit strictly it is not needed since we are queuing one call. But I will leave this up to you

Reply:
I'm gonna leave it. My thinking is that the backend logic should reflect what we support in the API, since either the frontend could change, or we may want to use the API directly in some strange customer support scenario. In all other cases where we could have >20 calls, we're using this, so it makes sense to be consistent.